### PR TITLE
Galaxy wrappers for the MFAssignR functions

### PR DESCRIPTION
**Pull request by KristinaGomoryova**
This PR includes galaxy wrappers for all the MFAssignR functions except for the IsoFiltR function, which was solved in PR #565.

Closes https://github.com/RECETOX/galaxytools/issues/473
Closes https://github.com/RECETOX/galaxytools/issues/472
Closes https://github.com/RECETOX/galaxytools/issues/470
Closes https://github.com/RECETOX/galaxytools/issues/469

### CODE REVIEW

File: tools/mfassignr/mfassignr_histnoise.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<tool id="mfassignr_histnoise" name="MFAssignR HistNoise" version="@TOOL_VERSION@+galaxy0">
```
Comment (bgruening) [inline tools/mfassignr/mfassignr_histnoise.xml:1]: please add a profile="" version

File: tools/mfassignr/mfassignr_histnoise.xml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    <description>Noise level assessment using the HistNoise.</description>
```
Comment (bgruening) [inline tools/mfassignr/mfassignr_histnoise.xml:2]:
```suggestion
    <description>Noise level assessment using the HistNoise</description>
```

File: tools/mfassignr/mfassignr_histnoise.xml (hunk 1 above)
Comment (bgruening) [reply tools/mfassignr/mfassignr_histnoise.xml:1]:
```suggestion
<tool id="mfassignr_histnoise" name="MFAssignR HistNoise" version="@TOOL_VERSION@+galaxy0" profile="23.0">
```

File: tools/mfassignr/mfassignr_histnoise.xml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        <import>macros.xml</import>
+        <import>help.xml</import>
+    </macros>
+    <edam_topics>
+        <edam_topic>topic_3172</edam_topic>    
+    </edam_topics>
+    <expand macro="creator" />
+    <expand macro="requirements" />
+    <command detect_errors="exit_code"><![CDATA[
+        Rscript -e 'source("${run_script}")'
```
Comment (bgruening) [inline tools/mfassignr/mfassignr_histnoise.xml:13]: Would something like that work as well? https://github.com/galaxyproject/tools-iuc/blob/6da96d865a3a557cfa3ad09e1cfa830519e73748/tools/ruvseq/ruvseq.xml#L46

So without source()

File: tools/mfassignr/mfassignr_histnoise.xml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    </configfiles>
+    <inputs>
+        <expand macro="histnoise_param"/>
+    </inputs>
+    <outputs>
+        <data name="Noise" format="txt" label="Noise level estimate by ${tool.name} on ${on_string}"/>
+        <data name="Hist_plot" format="png" label="Histogram plot by ${tool.name} on ${on_string}" from_work_dir="histplot.png"/>
+    </outputs>
+    <tests>
+        <test expect_failure="true"/>
```
Comment (bgruening) [inline tools/mfassignr/mfassignr_histnoise.xml:36]: here is something missing :)

File: tools/mfassignr/mfassignr_isofiltr.xml [hunk 5]
```diff
@@ -1,10 +1,14 @@
 <tool id="mfassignr_isofiltr" name="MFAssignR IsoFiltR" version="@TOOL_VERSION@+galaxy0">
-    <description>
-        IsoFiltR separates likely isotopic masses from monoisotopic masses in a mass list.
-    </description>
+    <description> Separates likely isotopic masses from monoisotopic masses in a mass list. </description>
```
Comment (bgruening) [inline tools/mfassignr/mfassignr_isofiltr.xml:2]:
```suggestion
    <description>Separates likely isotopic masses from monoisotopic masses in a mass list</description>
```

File: tools/mfassignr/mfassignr_kmdnoise.xml [hunk 6]
```diff
@@ -0,0 +1,53 @@
+<tool id="mfassignr_kmdnoise" name="MFAssignR KMDNoise" version="@TOOL_VERSION@+galaxy0">
+    <description>Noise level assessment using the KMDNoise.</description>
```
Comment (bgruening) [inline tools/mfassignr/mfassignr_kmdnoise.xml:2]:
```suggestion
    <description>Noise level assessment</description>
```

I guess those description can be shortend a bit. KMDNoise is already in the "name"

File: tools/mfassignr/mfassignr_kmdnoise.xml [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    </configfiles>
+    <inputs>
+        <expand macro="kmdnoise_param"/>
+    </inputs>
+    <outputs>
+        <data name="Noise" format="txt" label="Noise level estimate by ${tool.name} on ${on_string}"/>
+        <data name="KMD_plot" format="png" label="KMD plot by ${tool.name} on ${on_string}" from_work_dir="KMDplot.png"/>
+    </outputs>
+    <tests>
+        <test expect_failure="true"/>
```
Comment (bgruening) [inline tools/mfassignr/mfassignr_kmdnoise.xml:45]: :-)

File: tools/mfassignr/mfassignr_mfassign.xml [hunk 8]
```diff
@@ -0,0 +1,71 @@
+<tool id="mfassignr_mfassign" name="MFAssignR MFAssign" version="@TOOL_VERSION@+galaxy0">
+    <description>Molecular formula assignment.</description>
```
Comment (bgruening) [inline tools/mfassignr/mfassignr_mfassign.xml:2]:
```suggestion
    <description>Molecular formula assignment.</description>
```

File: tools/mfassignr/mfassignr_mfassign.xml [hunk 9]
```diff
@@ -0,0 +1,71 @@
+<tool id="mfassignr_mfassign" name="MFAssignR MFAssign" version="@TOOL_VERSION@+galaxy0">
```
Comment (bgruening) [inline tools/mfassignr/mfassignr_mfassign.xml:1]:
```suggestion
<tool id="mfassignr_mfassign" name="MFAssignR MFAssign" version="@TOOL_VERSION@+galaxy0" profile="23.0">
```

File: tools/mfassignr/mfassignr_mfassignCHO.xml [hunk 10]
```diff
@@ -0,0 +1,74 @@
+<tool id="mfassignr_mfassignCHO" name="MFAssignR MFAssignCHO" version="@TOOL_VERSION@+galaxy0">
+    <description>Molecular formula assignment (only with CHO).</description>
```
Comment (bgruening) [inline tools/mfassignr/mfassignr_mfassignCHO.xml:2]:
```suggestion
<tool id="mfassignr_mfassignCHO" name="MFAssignR MFAssignCHO" version="@TOOL_VERSION@+galaxy0" profile="23.0">
    <description>Molecular formula assignment (only with CHO)</description>
```

File: tools/mfassignr/mfassignr_recal.xml [hunk 11]
```diff
@@ -0,0 +1,75 @@
+<tool id="mfassignr_recal" name="MFAssignR Recal" version="@TOOL_VERSION@+galaxy0">
+    <description>Internal mass recalibration using a recalibrant series.</description>
```
Comment (bgruening) [inline tools/mfassignr/mfassignr_recal.xml:2]:
```suggestion
<tool id="mfassignr_recal" name="MFAssignR Recal" version="@TOOL_VERSION@+galaxy0" profile="23.0">
    <description>Internal mass recalibration using a recalibrant series</description>
```

File: tools/mfassignr/mfassignr_recallist.xml [hunk 12]
```diff
@@ -0,0 +1,44 @@
+<tool id="mfassignr_recallist" name="MFAssignR RecalList" version="@TOOL_VERSION@+galaxy0">
+    <description>Identification of candidate series for recalibration.</description>
```
Comment (bgruening) [inline tools/mfassignr/mfassignr_recallist.xml:2]:
```suggestion
<tool id="mfassignr_recallist" name="MFAssignR RecalList" version="@TOOL_VERSION@+galaxy0" profile="23.0">
    <description>Identification of candidate series for recalibration</description>
```

File: tools/mfassignr/mfassignr_snplot.xml [hunk 13]
```diff
@@ -0,0 +1,43 @@
+<tool id="mfassignr_snplot" name="MFAssignR SNplot" version="@TOOL_VERSION@+galaxy0">
+    <description>Noise level assessment using the SNplot function.</description>
```
Comment (bgruening) [inline tools/mfassignr/mfassignr_snplot.xml:2]:
```suggestion
<tool id="mfassignr_snplot" name="MFAssignR SNplot" version="@TOOL_VERSION@+galaxy0" profile="23.0">
    <description>Noise level assessment using the SNplot function</description>
```

File: tools/mfassignr/macros.xml [hunk 14]
```diff
@@ -26,6 +36,85 @@
...
+        <param name="input_file" type="data" format="tabular" label="Input data"
+               help= "Input data frame, first column is mass, second column is intensity"/>
+        <param name="SN" type="float" label="signal-to-noise threshold" value="0"
+               help= "A numeric value for situations where a predefined noise value is desired, default is 0"/>
+        <param name="bin" type="float" label="bindwidth of the histogram" value="0.01"
+               help= "A numeric value determining the binwidth of the histogram, default is 0.01"/>
+    </xml>
+
+    <xml name="noise_threshold_params">
+       <param name="sn_ratio" type="float" label="SN ratio" value="1"
```
Comment (bgruening) [inline tools/mfassignr/macros.xml:62]: value=1 but recommended is 6?

**Comment by bgruening** [comment]
Very cool @KristinaGomoryova!

File: tools/mfassignr/mfassignr_histnoise.xml (hunk 1 above)
Comment (KristinaGomoryova) [reply tools/mfassignr/mfassignr_histnoise.xml]: resolved in b8aa6958ebd2fa68c3f6dc0e15d0463e15f82c1e

File: tools/mfassignr/mfassignr_histnoise.xml (hunk 2 above)
Comment (KristinaGomoryova) [reply tools/mfassignr/mfassignr_histnoise.xml]: resolved in 2dbdd36efd69b7f996b258139927160bca954948

File: tools/mfassignr/mfassignr_histnoise.xml (hunk 3 above)
Comment (hechth) [reply tools/mfassignr/mfassignr_histnoise.xml]: Yes.

File: tools/mfassignr/macros.xml (hunk 14 above)
Comment (hechth) [reply tools/mfassignr/macros.xml:62]: Yeah - the tutorial recommends 6 but its not really explained anywhere why. Having a multiplier of 1 is like doing nothing, so seems neutral to me. If you think putting it by default to 6 is more intuitive we can change that.

Comment (bgruening) [reply tools/mfassignr/macros.xml:62]: The value and the help is just not aligned. So it should match IMHO... And I guess I would take whatever the documentation says.

File: tools/mfassignr/macros.xml [hunk 15]
```diff
@@ -26,6 +41,85 @@
...
     </xml>
 
+    <xml name="kmdnoise_param">
+        <param name="input_file" type="data" format="tabular" label="Input data"
+               help= "Input data frame, first column is mass, second column is intensity"/>
+        <param name="upper_y" type="float" label="upper limit for the y intercept" value="0.2"
+               help= "The upper y-intercept value to isolate noise peaks in the equation for the KMD plot: y = 0.001123*x + b. Default value is set to 0.2, so that it does not interact with any potentially double-charged peaks."/>
+        <param name="lower_y" type="float" label="lower limit for the y intercept" value="0.05"
+               help="The lower y-intercept value to isolate noise peaks in the equation for the KMD plot: y = 0.001123*x + b. Default value is set to 0.05 to ensure no analyte peaks are incorporated into the noise estimation."/>
+        <param name="upper_x" optional="true" type="float" label="upper limit for the x intercept"
```
[the remark this answers is not in the document]
Comment (hechth) [reply tools/mfassignr/macros.xml:46]: Not really - there don't seem to be any reasonable min max values.

File: tools/mfassignr/mfassignr_histnoise.xml (hunk 4 above)
Comment (hechth) [reply tools/mfassignr/mfassignr_histnoise.xml]: Tests are coming!

File: tools/mfassignr/mfassignr_recallist.xml [hunk 16]
```diff
@@ -35,7 +35,7 @@
     <tests>
         <test>
             <param name="input_file" value="mfassigncho/unambig.tabular" />
-            <output name="recal_series" file="recallist/recal_series.tabular" lines_diff="40"/>
+            <output name="recal_series" file="recallist/recal_series.tabular" lines_diff="100"/>
```
Comment (bgruening) [inline tools/mfassignr/mfassignr_recallist.xml:38]: use asserts here

File: tools/mfassignr/mfassignr_recallist.xml [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+
+            write.table(recal, file = '$recal_series', row.names= FALSE, sep = "\t")
+        ]]></configfile>
+    </configfiles>
+    <inputs>
+        <param name="input_file" type="data" format="tabular" label="Input data"
+               help= "Input data frame - output from MFAssign or MFAssignCHO. Recommended are 'Unambiguous' assignments."/>
+    </inputs>
+    <outputs>
+        <data name="recal_series" format="tabular" label="Recalibration series by ${tool.name} on ${on_string}"/>
```
Comment (bgruening) [inline tools/mfassignr/mfassignr_recallist.xml:33]: think about adding column names to your tabular files - will improve UX

Comment (hechth) [reply tools/mfassignr/mfassignr_recallist.xml:33]: The tools are quite specific and most inputs are outputs from other tools, so I don't expect people to come with their own files. Or do you mean something else than adding the colnames to the help text?

Comment (bgruening) [reply tools/mfassignr/mfassignr_recallist.xml:33]: Something like that: https://github.com/galaxyproject/tools-iuc/blob/048d2859041c0bcc4be4e904e637a96fe564bd36/tools/onto_toolkit/get_term_id_vs_term_name.xml#L17

Comment (hechth) [reply tools/mfassignr/mfassignr_recallist.xml:33]: This creates some very tight coupling between the package and the Galaxy tools :/

**Pull request merged**